pkg/templates/nstemplatetiers/nstemplatetier_generator_whitebox_test.go

```diff
@@ -182,76 +183,144 @@ func TestNewNSTemplateTier(t *testing.T) {
-						containsObj(t, ns.Template, fmt.Sprintf(`{"apiVersion":"networking.k8s.io/v1","kind":"NetworkPolicy","metadata":{"name":"allow-from-openshift-monitoring","namespace":"${USERNAME}-%s"},"spec":{"ingress":[{"from":[{"namespaceSelector":{"matchLabels":{"network.openshift.io/policy-group":"monitoring"}}}]}],"podSelector":{},"policyTypes":["Ingress"]}}`, ns.Type))
-
-						// All templates in the "team" tier and "-code" templates in other tiers should also have additional RoleBinding and Role
-						if kind == "code" || tier == "team" || tier == "advanced" {
-							require.Len(t, ns.Template.Objects, 8)
-							// Role & RoleBinding with additional permissions to edit roles/rolebindings
-							containsObj(t, ns.Template, fmt.Sprintf(`{"apiVersion":"rbac.authorization.k8s.io/v1","kind":"Role","metadata":{"name":"rbac-edit","namespace":"${USERNAME}-%s"},"rules":[{"apiGroups":["authorization.openshift.io","rbac.authorization.k8s.io"],"resources":["roles","rolebindings"],"verbs":["get","list","watch","create","update","patch","delete"]}]}`, ns.Type))
-							containsObj(t, ns.Template, fmt.Sprintf(`{"apiVersion":"rbac.authorization.k8s.io/v1","kind":"RoleBinding","metadata":{"name":"user-rbac-edit","namespace":"${USERNAME}-%s"},"roleRef":{"apiGroup":"rbac.authorization.k8s.io","kind":"Role","name":"rbac-edit"},"subjects":[{"kind":"User","name":"${USERNAME}"}]}`, ns.Type))
-						} else {
-							require.Len(t, ns.Template.Objects, 6)
+				t.Run(tier, func(t *testing.T) {
+					assert.Equal(t, namespace, actual.Namespace)
+					require.Len(t, actual.Spec.Namespaces, len(templatesByTier[tier].namespaceTemplates))
+					for kind, tmpl := range templatesByTier[tier].namespaceTemplates {
+						found := false
+						for _, ns := range actual.Spec.Namespaces {
+							if ns.Type != kind {
+								continue
+							}
+							found = true
+							assert.Equal(t, tmpl.revision, ns.Revision)
+							assertNamespaceTemplate(t, decoder, ns.Template, tier, ns.Type)
+							break
 						}
-						break
+						assert.True(t, found, "the namespace with the type wasn't found", "ns-type", kind)
 					}
-					assert.True(t, found, "the namespace with the type wasn't found", "ns-type", kind)
-				}
-				require.NotNil(t, actual.Spec.ClusterResources)
-				assert.Equal(t, templatesByTier[tier].clusterTemplate.revision, actual.Spec.ClusterResources.Revision)
-				cpuLimit := "1750m"
-				memoryLimit := "7Gi"
-				if tier == "team" {
-					cpuLimit = "2000m"
-					memoryLimit = "15Gi"
+					require.NotNil(t, actual.Spec.ClusterResources)
+					assert.Equal(t, templatesByTier[tier].clusterTemplate.revision, actual.Spec.ClusterResources.Revision)
+					assertClusterResourcesTemplate(t, decoder, actual.Spec.ClusterResources.Template, tier)
+				})
+			}
+		})
+
+		t.Run("with test assets", func(t *testing.T) {
```

Comment:
correct me if I'm wrong - this test, as well as the "failures" one, looks to be the same as the test on the lines 327-290. Is it just copy-paste or did you forget to commit some changes?

Reply:
this test verifies the content of the NSTemplateTiers, whereas in lines 327-390 we test the content of the TierTemplates. Sure, the tests are similar for now, because we have similar content in both resources, but this will change in the future.

Reply:
ok, I got it. I was looking at the wrong block of lines.
I refactored the tests in 885b1ee

Reply:
cool, thanks :+1: